Fix the shutdown timeout infinite loop bug and also fix the unreliable ordering bug in the instance redeploy command

lgp171188 opened the pull request:
This should also remove the unnecessary delays during the unit test runs.

UmanShahzad commented on instance/models/server.py:
```diff
@@ -461,15 +461,17 @@ def terminate(self):
         We don't have an explicit state for this and don't catch exceptions, which is why this is private.
         The caller is expected to handle any exceptions and retry if necessary.
         """
+        max_wait = max_wait if max_wait is not None else settings.SHUTDOWN_TIMEOUT
```
Might want to document that, I can imagine myself refactoring it if I didn't see the PR's context.

lgp171188 replied:
@UmanShahzad, I'll add a comment. :+1:

lgp171188 replied:
@UmanShahzad, I am not sure about adding a comment describing why testing the truthiness of `max_wait` shouldn't be done because it is just describing the way python works.

UmanShahzad replied:
Do the tests fail if this function disallows 0 with the refactoring I imagined above? If it does then it makes sense to not document. But if tests pass even if we perform the refactor, that could mean passing max_wait=0 in some places while not having intended effects, e.g. a hidden bug.

You can just leave this comment:

```python
# We purposely check for `None` rather than generically for falseness
# because it allows `max_wait = 0`.
```

lgp171188 replied:
> Do the tests fail if this function disallows 0 with the refactoring I imagined above? 
@UmanShahzad , the tests don't fail but each test loops for 10 minutes (the default value for the `SHUTDOWN_TIMEOUT`) before continuing and that wastes a lot of time. I will add the suggested comment.

UmanShahzad replied:
Thanks!

lgp171188 commented:
@UmanShahzad, I have updated the PR to include the comment and also added a trivial fix to use the official Ubuntu Xenial vagrant base box instead of the custom OpenCraft base box which was only meant to be a stop-gap measure.

UmanShahzad commented:
@lgp171188 Thanks, that should work. Did you test it? I can see a newcomer getting blocked if it ends up not working :P

UmanShahzad commented:
@lgp171188 Awesome, then lets merge (once tests pass).
